api/python/provisioner/commands/swupgrade.py

```diff
@@ -0,0 +1,147 @@
+from api.python.provisioner.salt import StatesApplier, local_minion_id
+from api.python.provisioner.salt_master import config_salt_master
+from api.python.provisioner.salt_minion import config_salt_minions
+from api.python.provisioner.vendor.attr import attr
+
+
+logger = logging.getLogger(__name__)
+
+
+COMPONENTS_FOR_UPGRADE = frozenset({'motr',
+                                    's3server',
+                                    'hare',
+                                    'ha.cortx-ha',
+                                    'sspl',
+                                    'uds',
+                                    'csm'})
+
+
+@attr.s(auto_attribs=True)
+class SWUpgrade(CommandParserFillerMixin):
+    input_type: Type[inputs.NoParams] = inputs.NoParams
+
+    @staticmethod
+    def _ensure_upgrade_repos_configuration():
+        logger.info("Ensure update repos are configured")
+        StatesApplier.apply(['components.misc_pkgs.swupgrade.repo'],
+                            local_minion_id())
+
+        # logger.info("Check yum repos are good")
+        # StatesApplier.apply(
+        #     ['components.misc_pkgs.sw_update.repo.sanity_check'],
+        #     local_minion_id()
+        # )
+
+    def run(self, targets):  # noqa: C901 FIXME
+        # logic based on https://jts.seagate.com/browse/EOS-6611?focusedCommentId=1833451&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-1833451  # noqa: E501
+
+        # TODO:
+        #   - create a state instead
+        #   - what about apt and other non-yum pkd managers
+        #   (downgrade is another more generic option but it requires
+        #    exploration of depednecies that are updated)
+        # TODO IMPROVE minions might be stopped here in case of rollback,
+        #      options: set up temp ssh config and rollback yum + minion config
+        #      via ssh as a fallback
+        rollback_ctx = None
+        try:
+            ensure_cluster_is_healthy()  # TODO: checker.run do that check too
+
+            checker = Check()
+            try:
+                check_res = checker.run(GroupChecks.SWUPDATE_CHECKS.value)
+            except Exception as e:
+                logger.warning("During pre-flight checks error happened: "
+                               f"{str(e)}")
+            else:
+                decision_maker = SWUpdateDecisionMaker()
+                decision_maker.make_decision(check_result=check_res)
+
+            self._ensure_upgrade_repos_configuration()
+
+            try:
+                _update_component('provisioner', targets)
+
+                # re-apply provisioner configuration to ensure
+                # that updated pillar is taken into account
+                _apply_provisioner_config(targets)
+
+                config_salt_master()
+
+                minion_conf_changes = config_salt_minions()
+
+                for component in COMPONENTS_FOR_UPGRADE:
+                    _update_component(component, local_minion_id())
```

Comment:
why we target only local minion id but use `targets` for provisioner? let's keep that generic for now even if have the scope of singlenode

Reply:
Done